### PR TITLE
Don't raise an error for malformed accept-encoding headers in compressed

### PR DESCRIPTION
**Pull request by mracos**
Related to #1427

### CODE REVIEW

File: src/cowboy_compress_h.erl [hunk 1]
```diff
@@ -72,21 +72,26 @@ early_error(StreamID, Reason, PartialReq, Resp, Opts) ->
...
+				%% equivalent to "gzip". (RFC7230 4.2.3)
+				case [E || E={<<"gzip">>, Q} <- Encodings, Q =/= 0] of
+					[] ->
+						#state{compress=undefined};
+					_ ->
+						#state{compress=gzip}
+				end
+		end
+  catch
+		%% Don't raise an error because of malformed or unsupported accept-encoding headers.
```
Comment (mracos) [inline src/cowboy_compress_h.erl]: I was not sure on what to put here, improvements appreciated 😄

File: src/cowboy_compress_h.erl [hunk 2]
```diff
@@ -72,21 +72,26 @@ early_error(StreamID, Reason, PartialReq, Resp, Opts) ->
...
-				_ ->
-					#state{compress=gzip}
-			end
+	try
+		case cowboy_req:parse_header(<<"accept-encoding">>, Req) of
+			%% Client doesn't support any compression algorithm.
+			undefined ->
+				#state{compress=undefined};
+			Encodings ->
+				%% We only support gzip so look for it specifically.
```
Comment (mracos) [inline src/cowboy_compress_h.erl]: I thought about doing this one too, wdyt?

File: src/cowboy_compress_h.erl [hunk 3]
```diff
@@ -72,21 +72,26 @@ early_error(StreamID, Reason, PartialReq, Resp, Opts) ->
...
-			%% @todo A recipient SHOULD consider "x-gzip" to be
-			%% equivalent to "gzip". (RFC7230 4.2.3)
-			case [E || E={<<"gzip">>, Q} <- Encodings, Q =/= 0] of
-				[] ->
-					#state{compress=undefined};
-				_ ->
-					#state{compress=gzip}
-			end
+	try
+		case cowboy_req:parse_header(<<"accept-encoding">>, Req) of
```
Comment (essen) [inline src/cowboy_compress_h.erl]: Don't wrap the case in a try, replace the case with the try. `try cowboy_req:parse... of ... catch`

File: src/cowboy_compress_h.erl (hunk 2 above)
Comment (essen) [reply src/cowboy_compress_h.erl]: No don't because that will change eventually.

File: src/cowboy_compress_h.erl (hunk 3 above)
Comment (mracos) [reply src/cowboy_compress_h.erl]: Nice one! [done here](https://github.com/ninenines/cowboy/compare/deb6dba33003528e934124c5d0827aa5d68eead7..8ffbe073c294743a5ce77ba2d637697b53306dcb).

**Comment by essen** [comment]
I'll try to get to this later today.

**Comment by essen** [comment]
Tweaked things and merged, thanks!

**Pull request closed without merging**